internal/initwd/module_install_test.go

```diff
@@ -739,10 +740,15 @@ func TestModuleInstaller_fromTests(t *testing.T) {
 	_, diags := inst.InstallModules(context.Background(), ".", "tests", false, false, hooks)
 	assertNoDiagnostics(t, diags)
 
+	// Use backslashes (\) for Windows paths in ModuleAddr
+	expectedModuleAddr := "test.tests.main.setup"
+	if runtime.GOOS == "windows" {
+		expectedModuleAddr = "test.tests\\main.setup"
```

Comment:
This is very strange, I wonder how `.` is being replaced with `\` in the module install hook

Reply:
I agree, I got stuck there looking in the code. Myabe it's the way the testdata`testdata/local-module-from-test` is laid out?

Reply:
Honestly, this seems like it might actually be a bug.  I'm digging into it a bit further.

Reply:
I'm looking at ModuleInstaller.installLocalModule and moduleInstallWalker, which leads me to configs.ModuleRequest.Path (joined with '.').

internal/configs/config_build.go:buildTestModules is probably where the req.Path is set and includes some logic around splitting on "/".  I'm guessing that's where the bug is.

Reply:
Yes you were correct. I've added a commit to this PR which uses filepath instead of path to solve that issue. Did a little test to show that `path.base()` always assumes the file system is UNIX:
```
var a_path_linux string = "bananas/main.tftest.hcl"
var a_path_windows string = "bananas\\main.tftest.hcl"

fmt.Println("The linux:", path.Base(a_path_linux))
fmt.Println("The windows: ", path.Base(a_path_windows))

fmt.Println("The linux filepath base:", filepath.Base(a_path_linux))
fmt.Println("The windows filepath base: ", filepath.Base(a_path_windows))

Output:
The linux: main.tftest.hcl
The windows:  bananas\main.tftest.hcl
The linux filepath base: main.tftest.hcl
The windows filepath base:  main.tftest.hcl
```